### ISSUE TITLE
Kubernetes: sidekiq issues

### ISSUE THREAD

**Issue by tamer-hassan** [bitnami/bitnami-docker-discourse#25]
**Description**

Running bitnami/discourse:latest (v1.8.3-dirty) in kubernetes 1.6.4 cluster on AWS.

According to: https://github.com/bitnami/bitnami-docker-discourse/blob/master/kubernetes.yml
discourse and sidekiq run in separate pods, and so there's a problem in heartbeat communication.
However, there is no problem running normal jobs in queue like sending out mails etc.

**Steps to reproduce the issue:**

1. kubectl apply -f https://github.com/bitnami/bitnami-docker-discourse/blob/master/kubernetes.yml
    (or a stripped down version of it if you already have postgres and/or redis elsewhere, in which case set ENV variables for them in both discourse and discourse-sidekiq deployments)
2. Check https://discourse.example.com/sidekiq/queues/critical
3. Notice many of these jobs piling up:
`Jobs::RunHeartbeat           |            [{"current_site_id"=>"default"}]`
4. Also notice that attempting to delete any of these jobs results in a "Forbidden" page.

**Comment by thassan** [comment]
To answer myself on the two issues (steps 3 and 4 above)
** The heartbeat issue **
From looking at https://discourse.mysite.com/logs I noticed the Job Exception with count over 1300, and in the env tab of that error was:
> current_hostname | www.example.com

So I edited (on the persistent volume) discourse-sidekiq/conf/discourse.conf and added:
`hostname = discourse.mysite.com`
and restarted the sidekiq pod/instance.
This solved the heartbeat issue I believe, from looking at discourse.mysite.com/sidekiq/scheduler it looks like heartbeat is running ok now.

**The Forbidden error**

When I click an action (delete/trigger/etc.), I get forbidden error, and at the same time in /opt/bitnami/discourse/logs/production.log

> App 2012 stderr: W, [2017-07-13T04:46:02.841195 #2043]  WARN -- : attack prevented by Rack::Protection::HttpOrigin

From https://github.com/mperham/sidekiq/wiki/Monitoring#forbidden
> Tip: If you have a load balancer -> reverse proxy -> web application server setup, make sure you're setting the X-Forwarded-Proto header properly. If your load balancer is your SSL termination point you need to make sure it's set to the value forwarded from the load balancer, not the internal value (otherwise rack-protection will see the request as coming via http, not https).

In our case, CloudFront is doing the SSL termination and using an HTTP-only ELB as origin.
So, according to https://github.com/mperham/sidekiq/issues/2560#issuecomment-245229685
I need to add this to nginx conf , at least for the location /sidekiq:
`proxy_set_header X-Forwarded-Proto https;`

But nginx.conf is not on persistent storage.
hopefully this could be solved in next release?

**Comment by thassan** [comment]
Sidekiq crashed, and no logs in stdout. Had to go inside the container and find out why it stopped.

Apparently the plugin discourse-nginx-performance-report is looking for /var/log/nginx/access.log and not finding it.

```
jasmin@gs60:~$ kubectl logs sidekiq-1803692988-mxc82

Welcome to the Bitnami discourse container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-discourse
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-discourse/issues
Send us your feedback at containers@bitnami.com

nami    INFO  Initializing discourse-sidekiq
nami    INFO  discourse-sidekiq successfully initialized
Starting discourse-sidekiq...
# Logfile created on 2017-07-13 04:11:48 +0000 by logger.rb/56438
2017-07-13T04:11:53.076Z 60 TID-gpwarhgjs INFO: Booting Sidekiq 4.2.4 with redis options {:host=>"redis-service", :port=>6379, :namespace=>"sidekiq", :url=>nil}

jasmin@gs60:~$ kubectl exec -ti sidekiq-1803692988-mxc82 bash
root@sidekiq-1803692988-mxc82:/# cd /opt/bitnami/discourse-sidekiq/                                                    
root@sidekiq-1803692988-mxc82:/opt/bitnami/discourse-sidekiq# tail log/production.log 
root@sidekiq-1803692988-mxc82:/opt/bitnami/discourse-sidekiq# tail log/sidekiq.log    
# Logfile created on 2017-07-13 04:11:48 +0000 by logger.rb/56438
2017-07-13T04:11:53.076Z 60 TID-gpwarhgjs INFO: Booting Sidekiq 4.2.4 with redis options {:host=>"redis-service", :port=>6379, :namespace=>"sidekiq", :url=>nil}
/opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/lib/log_analyzer.rb:127:in `initialize': No such file or directory @ rb_sysopen - /var/log/nginx/access.log (Errno::ENOENT)
	from /opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/lib/log_analyzer.rb:127:in `open'
	from /opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/lib/log_analyzer.rb:127:in `block in analyze'
	from /opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/lib/log_analyzer.rb:126:in `each'
	from /opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/lib/log_analyzer.rb:126:in `analyze'
	from /opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/lib/log_analyzer.rb:60:in `analyze'
	from /opt/bitnami/discourse-sidekiq/plugins/discourse-nginx-performance-report/script/nginx_analyze.rb:14:in `<main>'
root@sidekiq-1803692988-mxc82:/opt/bitnami/discourse-sidekiq#
```

**Comment by chipironcin** [comment]
Hi @jasminhassan 
Sorry for the delay, my team is just catching up with all the Github issues at our organization.

I can confirm your issues and I opened an internal task to fix them as they can only be fixed during the generation of the source files. We will let you know in this thread when the issues are resolved.

Many thanks for sharing the fixes publicly. Other people will benefit from them in the meantime 👏

**Comment by beltran-rubo** [comment]
Could you check the latest released version of Discourse "2.0.3-r11" or later? We already changed the configuration for Sidekiq on order to be able to process jobs from the queues.
